tools/test/examples/update.py

```diff
@@ -149,52 +197,100 @@ def prepare_fork(arm_example):
     
     """
 
-    print "In " + os.getcwd()
+    update_log.debug("In: ", os.getcwd())
 
     for cmd in [['git', 'remote', 'add', 'armmbed', arm_example],
                 ['git', 'fetch', 'armmbed'],
                 ['git', 'reset', '--hard', 'armmbed/master'],
                 ['git', 'push', '-f', 'origin']]:
         if run_cmd(cmd):
-            print("preparation of the fork failed!")
+            update_log.error("Fork preparation failed")
             return False
     return True
 
+def prepare_branch(branch):
+
+    update_log.debug("Preparing branch: %s", branch)
+
+    # Check if branch already exists or not.
+    cmd = ['git', 'branch']
+    return_code, output = run_cmd_with_output(cmd)
+
+    if not branch in output:
+        # OOB branch does not exist thus create it and then check it out
+        cmd = ['git', 'checkout', '-b', branch]
+        return_code = run_cmd(cmd)
+        if not return_code:
```

Comment:
You've done such a nice job with handling errors all over this script, do you want to handle the failure case here as well?